### PR TITLE
PLATUI-2755: sbt-accessibility-linter supports Play 2.9 and Play 3.0

### PR DESCRIPTION
**Pull request by kyle-bowden**

### CODE REVIEW

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Chosen option: ???
+
+### Positive Consequences
+
+### Negative Consequences
+
+## Pros and Cons of the Options
+
+### Option 1: Specific Plugin Version for Each Play Version
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:37]: What about
```suggestion
### Option 1: Create play-specific plugin versions - using code gen to inject the right scalatest library version
```

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* Good, because it avoids the need for a multi-project build.
+* Bad, because it requires separate builds for each new Play version, increasing maintenance.
+
+### Option 2: Reflection to Determine Play Version
+
+* Good, because it eliminates the need for different plugin versions.
+* Good, because it's relatively easy to implement.
+* Bad, because reflection can lead to unexpected edge cases in the JVM environment.
+
+### Option 3: Macros for Compile-Time Play Version Detection
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:48]: What about
```suggestion
### Option 3: Create play-specific plugin versions - using macros to inject the right scalatest library versions
```
I think this is right - that we'd be expanding the macros at compile time of the plugin (not of the service) - but we'd still need multiple versions of the library

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+* Good, because it identifies the Play version at compile time.
+* Bad, because macro usage can be complex and challenging to maintain.
+
+### Option 4: Decouple scalatest Library from Plugin
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:53]:
```suggestion
### Option 4: Make users add the right scalatest library for their version of play themselves
```
I think decouple implies a bit more than we're doing - I don't think we'd really be decoupling stuff with this, just making integration more complicated

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* Maintainability of the plugin
+* Ease of implementation
+* Future-proofing against new Play Framework versions
+
+## Considered Options
+
+* Option 1: Build a specific version of the `sbt-accessibility-linter` plugin for each Play Framework version.
+* Option 2: Use reflection in the `sbt-accessibility-linter` to determine the consuming project's Play version and dynamically include the correct `scalatest-accessibility-linter` library.
+* Option 3: Employ macros to determine the Play version at compile time and include the appropriate `scalatest-accessibility-linter` library.
+* Option 4: Decouple the `scalatest-accessibility-linter` library from the `sbt-accessibility-linter` plugin.
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:25]:
```suggestion
* Option 4: Decouple the `scalatest-accessibility-linter` library from the `sbt-accessibility-linter` plugin.
* Option 5: Only support one version of play at a time - so future updates for old play versions would have to be done as patch versions
```
I think in the future I'd probably say do option 5 if we're not sure about future of something - like we did for play-ui

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+### Option 1: Specific Plugin Version for Each Play Version
+
+* Good, because it avoids the need for a multi-project build.
+* Bad, because it requires separate builds for each new Play version, increasing maintenance.
+
+### Option 2: Reflection to Determine Play Version
+
+* Good, because it eliminates the need for different plugin versions.
+* Good, because it's relatively easy to implement.
+* Bad, because reflection can lead to unexpected edge cases in the JVM environment.
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:46]:
```suggestion
* Bad, because reflection can lead to unexpected edge cases in the JVM environment. But because this is in test scope, and the tests aren't necessary to have they're just a nice to have, it's easy for teams to drop if there are any problems
```

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+### Option 1: Specific Plugin Version for Each Play Version
+
+* Good, because it avoids the need for a multi-project build.
+* Bad, because it requires separate builds for each new Play version, increasing maintenance.
+
+### Option 2: Reflection to Determine Play Version
+
+* Good, because it eliminates the need for different plugin versions.
+* Good, because it's relatively easy to implement.
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:45]:
```suggestion
* Good, because it's relatively easy to implement. We already have an implementation with some test coverage contributed from @wolfendale
```

File: docs/adr/0001-compatibility-with-different-play-version.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+* Bad, because reflection can lead to unexpected edge cases in the JVM environment.
+
+### Option 3: Macros for Compile-Time Play Version Detection
+
+* Good, because it identifies the Play version at compile time.
```
Comment (oscarduignan) [inline docs/adr/0001-compatibility-with-different-play-version.md:50]:
```suggestion
* Good, because it fits with the existing use of PLAY_VERSION to identify different play versions - rather than having multiple ways (codegen)
* Good, because we wouldn't risk any runtime problems like have a small risk of with reflection
```

**Review by TimothyFothergill — APPROVED** [review]
lgtm
